avogadro/qtgui/interfacescript.cpp

```diff
@@ -145,7 +161,74 @@ void InputGenerator::reset()
   m_highlightStyles.clear();
 }
 
-bool InputGenerator::generateInput(const QJsonObject &options_,
+bool InterfaceScript::runWorkflow(const QJsonObject &options_,
+                                  Core::Molecule &mol)
+{
+  m_errors.clear();
+  m_warnings.clear();
+  m_filenames.clear();
+  qDeleteAll(m_fileHighlighters.values());
+  m_fileHighlighters.clear();
+  m_mainFileName.clear();
+  m_files.clear();
+
+  // Add the molecule file to the options
+  QJsonObject allOptions(options_);
+  if (!insertMolecule(allOptions, mol))
+    return false;
+
+  QByteArray json(m_interpreter->execute(QStringList() << "--run-workflow",
+                                         QJsonDocument(allOptions).toJson()));
+
+  if (m_interpreter->hasErrors()) {
+    m_errors << m_interpreter->errorList();
+    return false;
+  }
+
+  QJsonDocument doc;
+  if (!parseJson(json, doc))
+    return false;
+
+  // Update cache
+  bool result = true;
+  if (doc.isObject()) {
+    QJsonObject obj = doc.object();
+
+    // Check for any warnings:
+    if (obj.contains("warnings")) {
+      if (obj["warnings"].isArray()) {
+        foreach (const QJsonValue &warning, obj["warnings"].toArray()) {
+          if (warning.isString())
+            m_warnings << warning.toString();
+          else
+            m_errors << tr("Non-string warning returned.");
+        }
+      }
+      else {
+        m_errors << tr("'warnings' member is not an array.");
+      }
+    }
+
+    // TODO: add undo / redo and smart updates
+    Io::FileFormatManager &formats = Io::FileFormatManager::instance();
+    QScopedPointer<Io::FileFormat> format(formats.newFormatFromFileExtension(
+                                            "cjson"));
+    // convert the "cjson" field to a string
+    QJsonObject cjsonObj = obj["cjson"].toObject();
+    QJsonDocument doc(cjsonObj);
+    QString strCJSON(doc.toJson(QJsonDocument::Compact));
+    if (!strCJSON.isEmpty()) {
+      result = format->readString(strCJSON.toStdString(), mol);
+      // TODO: need to indicate changes to the QtGui Molecule
```

Comment:
mol is a Core::Molecule as a parameter, and I cannot seem to cast to a QtGui::Molecule here - I always get NULL.

Ideas or fixes are *very* welcome